integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonOutputFormat.java

```diff
@@ -61,24 +68,32 @@ public void checkOutputSpecs(FileSystem fileSystem, JobConf jobConf) throws IOEx
   public FileSinkOperator.RecordWriter getHiveRecordWriter(JobConf jc, Path finalOutPath,
       Class<? extends Writable> valueClass, boolean isCompressed, Properties tableProperties,
       Progressable progress) throws IOException {
+    ThreadLocalSessionInfo.setConfigurationToCurrentThread(jc);
     CarbonLoadModel carbonLoadModel = null;
+    // Try to get loadmodel from JobConf.
     String encodedString = jc.get(LOAD_MODEL);
     if (encodedString != null) {
       carbonLoadModel =
           (CarbonLoadModel) ObjectSerializationUtil.convertStringToObject(encodedString);
-    }
-    if (carbonLoadModel == null) {
-      carbonLoadModel = HiveCarbonUtil.getCarbonLoadModel(tableProperties, jc);
     } else {
-      for (Map.Entry<Object, Object> entry : tableProperties.entrySet()) {
-        carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable().getTableInfo().getFactTable()
-            .getTableProperties().put(entry.getKey().toString().toLowerCase(),
-            entry.getValue().toString().toLowerCase());
+      // Try to get loadmodel from Container environment.
+      encodedString = System.getenv("carbon");
+      if (encodedString != null) {
+        carbonLoadModel =
+            (CarbonLoadModel) ObjectSerializationUtil.convertStringToObject(encodedString);
+      } else {
+        carbonLoadModel = HiveCarbonUtil.getCarbonLoadModel(tableProperties, jc);
       }
     }
+    for (Map.Entry<Object, Object> entry : tableProperties.entrySet()) {
```

Comment:
where are these table properties validated ?

Reply:
in CarbonLoadModelBuilder

Reply:
ok

Reply:
CarbonLoadModelBuilder is used to create LoadModel before this. So the properties would be validated